internal/storage/v1/elasticsearch/factory_test.go

```diff
@@ -214,41 +133,97 @@ func TestTagKeysAsFields(t *testing.T) {
+		},
+		{
+			spanTemplateService: func() *mocks.TemplateCreateService {
+				tService := &mocks.TemplateCreateService{}
+				tService.On("Body", mock.Anything).Return(tService)
+				tService.On("Do", context.Background()).Return(nil, nil)
+				return tService
+			},
+			serviceTemplateService: func() *mocks.TemplateCreateService {
+				tService := &mocks.TemplateCreateService{}
+				tService.On("Body", mock.Anything).Return(tService)
+				tService.On("Do", context.Background()).Return(nil, nil)
+				return tService
+			},
+			indexPrefix: "test",
+		},
+		{
+			err: "span-template-error",
+			spanTemplateService: func() *mocks.TemplateCreateService {
+				tService := new(mocks.TemplateCreateService)
+				tService.On("Body", mock.Anything).Return(tService)
+				tService.On("Do", context.Background()).Return(nil, errors.New("span-template-error"))
+				return tService
+			},
+			serviceTemplateService: func() *mocks.TemplateCreateService {
+				tService := new(mocks.TemplateCreateService)
+				tService.On("Body", mock.Anything).Return(tService)
+				tService.On("Do", context.Background()).Return(nil, nil)
+				return tService
+			},
+		},
+		{
+			err: "service-template-error",
+			spanTemplateService: func() *mocks.TemplateCreateService {
+				tService := new(mocks.TemplateCreateService)
+				tService.On("Body", mock.Anything).Return(tService)
+				tService.On("Do", context.Background()).Return(nil, nil)
+				return tService
+			},
+			serviceTemplateService: func() *mocks.TemplateCreateService {
+				tService := new(mocks.TemplateCreateService)
+				tService.On("Body", mock.Anything).Return(tService)
+				tService.On("Do", context.Background()).Return(nil, errors.New("service-template-error"))
+				return tService
+			},
+		},
+	}
 
-func TestConfigureFromOptions(t *testing.T) {
-	f := NewFactory()
-	o := &Options{
-		Config: namespaceConfig{Configuration: escfg.Configuration{Servers: []string{"server"}}},
+	for _, test := range tests {
+		f := FactoryBase{}
+		mockClient := &mocks.Client{}
+		f.newClientFn = func(_ *escfg.Configuration, _ *zap.Logger, _ metrics.Factory) (es.Client, error) {
+			return mockClient, nil
+		}
+		f.logger = zaptest.NewLogger(t)
+		f.metricsFactory = metrics.NullFactory
+		f.config = &escfg.Configuration{CreateIndexTemplates: true, Indices: escfg.Indices{IndexPrefix: test.indexPrefix}}
+		f.tracer = otel.GetTracerProvider()
+		client, err := f.newClientFn(&escfg.Configuration{}, zaptest.NewLogger(t), metrics.NullFactory)
+		require.NoError(t, err)
+		f.client.Store(&client)
+		f.templateBuilder = es.TextTemplateBuilder{}
+		jaegerSpanId := test.indexPrefix.Apply("jaeger-span")
+		jaegerServiceId := test.indexPrefix.Apply("jaeger-service")
+		mockClient.On("CreateTemplate", jaegerSpanId).Return(test.spanTemplateService())
+		mockClient.On("CreateTemplate", jaegerServiceId).Return(test.serviceTemplateService())
+		err = f.createTemplates(context.Background())
+		if test.err != "" {
+			require.Error(t, err, test.err)
```

Comment:
the argument `test.err` here is used as a log message. You want ErrorContains check.